Tests: Fix gating tests for 9.2

mmatsuya opened the pull request:
It fixes test from tire1_2 that is failling in gating

1. src/tests/multihost/alltests/test_automount.py   there is issue with autofs email thead: [CRASH] prep Package: autofs-1:5.1.7-36.el9
2. src/tests/multihost/alltests/test_automount_from_bash.py test did not rised error as last cd - command was successful, so i have remove cd - part(/folder1/folder2/projects does not exists)
3. src/tests/multihost/alltests/test_ldap_password_policy.py  password provied was wrong.
4. src/tests/multihost/alltests/test_backtrace.py ---  need to modify this test as per current log format

(cherry picked from 2096f45527d4513ae52547fafd383bd2542d7f79)

aborah-sudo commented:
As cherry picked. LGTM

shridhargadekar reviewed (changes requested):
Apart from mentioned comment, rest looks good.

mmatsuya commented:
Hi @shridhargadekar 
I modified the range(). I found the same usage of range(1, N) in other locations of test_automount.py and test_automount_from_bash.py, so I modified them as well.

pbrezina commented:
Pushed PR: https://github.com/SSSD/sssd/pull/7015

* `sssd-2-8`
    * f28c6236969cf5c51871f24771009e26b6fb8f7e - Tests: Fix gating tests for 9.2
